src/components/SceneApp/Queries/ClusterMappingQueries.ts

```diff
@@ -77,9 +77,20 @@ const transformTrendData: (clusterData: SceneQueryRunner) => CustomTransformOper
         );
 }
 
+export const GetClusterToSubscription = (clusterData: SceneQueryRunner) => {
+    const clusters = clusterData.state.data?.series.find((s) => s.refId === "clusters")?.fields[0]?.values ?? [];
+    const subscriptionIds = clusterData.state.data?.series.find((s) => s.refId === "clusters")?.fields.find((f) => f.name === "subscriptionId")?.values ?? [];
+    const clusterToSubscription = new Map<string, string>();
+    for (const cluster of clusters) {
+        clusterToSubscription.set(cluster, subscriptionIds[clusters.indexOf(cluster)]);
+    }
+    return clusterToSubscription;
+}
```

Comment:
can we store this as part of clusterMappings and read from that instead? https://github.com/microsoft/AzureMonitor-Grafana/blob/3d49cd76b2527a4893ce8f7d773a70d9167f0135/src/components/SceneApp/Queries/queryUtil.ts#L115C7-L122C8

Reply:
What would be the benefit of doing that?

Reply:
We keep everything in one place. This is essentially doing the same as `clusterMappings`, keeping a map of cluster -> subId, and `clusterMappings` also does that but keeps a map of cluster -> multiple properties.

Additionally, all scenes already have access to `clusterMappings`, so if down the line we need access of cluster to subId, we'd be doing one calculation (just for `clusterMappings`) instead of two (one for `clusterMappings` and one for `clusterToSubscription`)

Reply:
The data will not be resolved by then that is why we need to pass the scene query runner to the transform.